### PR TITLE
Rename type -> id

### PR DESCRIPTION
**Pull request by HenryNguyen5**
This PR contains a commit based on https://github.com/smartcontractkit/chainlink/pull/13061 which will be rebased out when approved.

### CODE REVIEW

File: contracts/src/v0.8/keystone/CapabilityRegistry.sol [hunk 1]
```diff
@@ -183,19 +184,20 @@ contract CapabilityRegistry is OwnerIsCreator, TypeAndVersionInterface {
...
+  /// @notice This function returns a Capability by its compressed ID. Use `getCompressedCapabilityID` to get the compressed ID.
+  function getCapability(bytes32 compressedId) public view returns (Capability memory) {
+    return s_capabilities[compressedId];
   }
 
   /// @notice This functions returns a Capability ID packed into a bytes32 for cheaper access
   /// @return bytes32 A unique identifier for the capability
-  function getCapabilityID(bytes32 capabilityType, bytes32 version) public pure returns (bytes32) {
-    return keccak256(abi.encodePacked(capabilityType, version));
+  function getCompressedCapabilityID(bytes32 labeledName, bytes32 version) public pure returns (bytes32) {
```
Comment (bolekk) [inline contracts/src/v0.8/keystone/CapabilityRegistry.sol:200]: Is it really a reduction from 64 bytes down to 32? If we are so close then maybe it's just possible to require all IDs to fit in 32 bytes? I think that could simplify things quite a bit. Can we try to come up with some reasonable limits on labels or whatever takes most space?

**Review by bolekk — CHANGES_REQUESTED** [review]

File: contracts/src/v0.8/keystone/CapabilityRegistry.sol (hunk 1 above)
Comment (DeividasK) [reply contracts/src/v0.8/keystone/CapabilityRegistry.sol]: > Is it really a reduction from 64 bytes down to 32?
We take in two values with bytes32 and return bytes32, so it is a reduction to 32 bytes.

> If we are so close then maybe it's just possible to require all IDs to fit in 32 bytes?
I don't think that's realistic. Assuming the simplest version `XX.XX.XX` it already takes up 8 bytes. `read_chain-ethereum:mainnet` would not fit into the 24 remaining bytes (it is 27 bytes long). 32 bytes for capability type is restrictive as it is. If anything, we might want to consider expanding it to allow things like scope, e.g., `@chainlink` prefix.

File: core/services/workflows/models_yaml.go [hunk 2]
```diff
@@ -248,88 +250,88 @@ type stepDefinitionYaml struct {
...
 		Config: s.Config,
 	}
 }
 
-// stepDefinitionType represents both the string and table representations of the "type" field in a stepDefinition.
-type stepDefinitionType struct {
-	typeStr   string
-	typeTable *stepDefinitionTableType
+// stepDefinitionId represents both the string and table representations of the "id" field in a stepDefinition.
+type stepDefinitionId struct {
```
Comment (jmank88) [inline core/services/workflows/models_yaml.go:260]:
```suggestion
type stepDefinitionID struct {
```

**Comment by HenryNguyen5** [comment]
@DeividasK @cedric-cordenier @bolekk @cds95 

I'm using the term "compressedCapabilityID" here, but it's a bit of a misnomer, although it does describe why we want to use a hash of the ID (compression purposes).

 I'd also be happy to rename to`hashedCapabilityID` if there's preference for that. I'm aware that this is quite wordy, we could have some shortforms of this naming too. I've listed some alternatives below. Please vote for your preference.

- :hamburger: for `compressedCapabilityID`
- :hotdog: for `hashedCapabilityID`
- :pizza: for `compressedCID`
- :fries: for `hashedCID`

**Comment by DeividasK** [comment]
> @DeividasK @cedric-cordenier @bolekk @cds95
> 
> I'm using the term "compressedCapabilityID" here, but it's a bit of a misnomer, although it does describe why we want to use a hash of the ID (compression purposes).
> 
> I'd also be happy to rename to`hashedCapabilityID` if there's preference for that. I'm aware that this is quite wordy, we could have some shortforms of this naming too. I've listed some alternatives below. Please vote on this comment for your preference.
> 
> * 🚀 for `compressedCapabilityID`
> * ❤️ for `hashedCapabilityID`
> * 👀 for `compressedCID`
> * 🎉 for `hashedCID`

CID conflicts with Workflow CID (Content ID), so I'd prefer the longer terms. +1 for `hashedCapabilityID`. Having a "normal" Capability ID and "hashed" Capability ID seems weird. Alternatively, we could stick with Hashed Capability ID being _the_ Capability ID and internally just use labelledName and version when we need to refer to non-hashed capability ID 🤷‍♂️

File: contracts/src/v0.8/keystone/CapabilityRegistry.sol (hunk 1 above)
Comment (bolekk) [reply contracts/src/v0.8/keystone/CapabilityRegistry.sol]: That's a bit sad but OK, if we need more then let's use a compressed name separately.

**Review by bolekk — APPROVED** [review]

**Review by bolekk — APPROVED** [review]

**Pull request merged**